### PR TITLE
MC-13977: create delivery domains

### PR DESCRIPTION
**Pull request by anusha-dey**
#### Changes made

- create delivery domains

#### Related PRs
- https://github.com/cloudops/cloudmc-stackpath-plugin/pull/161

### CODE REVIEW

File: source/includes/stackpath/_delivery_domains.md [hunk 1]
```diff
@@ -117,4 +117,50 @@ Delete a delivery domain of a site in a given [environment](#administration-envi
...
+    -d "request_body" \
+    "https://cloudmc_endpoint/api/v1/services/stackpath/test-area/deliverydomains?siteId=a2fefb15-7e31-4c72-87e0-5a892e91c8d9&operation=Create"
+```
+
+> Request body example:
+
+```json
+{
+  "domain": "new-domain.com",
+  "validatedAt": "2021-02-26T19:00:15.177411Z"
```
Comment (fyrnaga) [inline source/includes/stackpath/_delivery_domains.md]: This should not be provided on a create

File: source/includes/stackpath/_delivery_domains.md [hunk 2]
```diff
@@ -117,4 +117,50 @@ Delete a delivery domain of a site in a given [environment](#administration-envi
...
+
+Create a delivery domain of a site in a given [environment](#administration-environments).
+
+Required | &nbsp;
+---------- | -----------
+`domain`<br/>*String* | The site's domain name.
+
+Optional | &nbsp;
+---------- | -----------
+`validatedAt`<br/>*String* | The date the domain was validated to be pointing to Stackpath.
```
Comment (fyrnaga) [inline source/includes/stackpath/_delivery_domains.md]: Is this required for the creation? The API does not document it

Comment (GuillaumeVialle) [reply source/includes/stackpath/_delivery_domains.md]: This is a value returned and should not be provided

Comment (anusha-dey) [reply source/includes/stackpath/_delivery_domains.md]: removed https://github.com/cloudops/cloudmc-api-docs/pull/317/commits/5ecc01b84a7d8218080b49460ff1e72977b94125

File: source/includes/stackpath/_delivery_domains.md (hunk 1 above)
Comment (anusha-dey) [reply source/includes/stackpath/_delivery_domains.md]: removed https://github.com/cloudops/cloudmc-api-docs/pull/317/commits/5ecc01b84a7d8218080b49460ff1e72977b94125

**Review by fyrnaga — APPROVED** [review]
lgtm

File: source/includes/stackpath/_delivery_domains.md [hunk 3]
```diff
@@ -117,4 +117,45 @@ Delete a delivery domain of a site in a given [environment](#administration-envi
...
+
+<!-------------------- CREATE A DELIVERY DOMAIN -------------------->
+
+### Create a delivery domain
+
+```shell
+curl -X POST \
+    -H "MC-Api-Key: your_api_key" \
+    -d "request_body" \
+    "https://cloudmc_endpoint/api/v1/services/stackpath/test-area/deliverydomains?siteId=a2fefb15-7e31-4c72-87e0-5a892e91c8d9&operation=Create"
```
Comment (GuillaumeVialle) [inline source/includes/stackpath/_delivery_domains.md]: I don't think the `&operation=Create` is necessary at the end.

File: source/includes/stackpath/_delivery_domains.md [hunk 4]
```diff
@@ -117,4 +117,45 @@ Delete a delivery domain of a site in a given [environment](#administration-envi
...
+> The above command returns a JSON structured like this:
+
+```json
+{
+  "taskId": "57fc8d89-6f13-451b-8b66-fcd96e1fedbd",
+  "taskStatus": "SUCCESS"
+}
+```
+
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/deliverydomains?siteId=:id&operation=Create</code>
```
Comment (GuillaumeVialle) [inline source/includes/stackpath/_delivery_domains.md]: Same here, I think you can remove `&operation=Create`

File: source/includes/stackpath/_delivery_domains.md [hunk 5]
```diff
@@ -117,4 +117,45 @@ Delete a delivery domain of a site in a given [environment](#administration-envi
...
+
+Create a delivery domain of a site in a given [environment](#administration-environments).
+
+Required | &nbsp;
+---------- | -----------
+`domain`<br/>*String* | The site's domain name.
+
+Attributes | &nbsp;
+------- | -----------
+`taskId` <br/>*string* | The task id related to the site creation.
```
Comment (GuillaumeVialle) [inline source/includes/stackpath/_delivery_domains.md]: `site` -> `domain`

File: source/includes/stackpath/_delivery_domains.md (hunk 3 above)
Comment (anusha-dey) [reply source/includes/stackpath/_delivery_domains.md]: removed 0db866c

File: source/includes/stackpath/_delivery_domains.md (hunk 4 above)
Comment (anusha-dey) [reply source/includes/stackpath/_delivery_domains.md]: removed 0db866c

File: source/includes/stackpath/_delivery_domains.md (hunk 5 above)
Comment (anusha-dey) [reply source/includes/stackpath/_delivery_domains.md]: 0db866c

**Review by GuillaumeVialle — APPROVED** [review]
Thanks for the changes

**Pull request merged**